filebeat/input/logv2/convert_test.go

```diff
@@ -0,0 +1,443 @@
+		  ],
+		  "prospector": {
+		    "scanner": {
+		      "fingerprint": {
+		        "enabled": false
+		      }
+		    }
+		  },
+		  "take_over": {
+		    "enabled": true
+		  },
+		  "type": "filestream"
+		}
+		`,
+		},
+
+		"parsers are added after json": {
+			logYamlCfg: `
+id: foo
+paths:
+  - /tmp/foo
+json.expand_keys: true
+parsers:
+  - container: ~
+`,
+			expectedJsonCfg: `
+		{
+		  "file_identity": {
+		    "native": null
+		  },
+		  "id": "foo",
+          "parsers": [
+            {
+              "ndjson": {
+                "expand_keys": true,
+                "target": "json"
+              }
+            },
+            {
+              "container": null
+            }
+          ],
+		  "paths": [
+		    "/tmp/foo"
+		  ],
+		  "prospector": {
+		    "scanner": {
+		      "fingerprint": {
+		        "enabled": false
+		      }
+		    }
+		  },
+		  "take_over": {
+		    "enabled": true
+		  },
+		  "type": "filestream"
+		}
+		`,
+		},
+
+		"parsers are added after multiine": {
+			logYamlCfg: `
+id: foo
+paths:
+  - /tmp/foo
+multiline.type: count
+parsers:
+  - container: ~
+`,
+			expectedJsonCfg: `
+		{
+		  "file_identity": {
+		    "native": null
+		  },
+		  "id": "foo",
+          "parsers": [
+            {
+              "multiline": {
+                "type": "count"
+              }
+            },
+            {
+              "container": null
+            }
+          ],
+		  "paths": [
+		    "/tmp/foo"
+		  ],
+		  "prospector": {
+		    "scanner": {
+		      "fingerprint": {
+		        "enabled": false
+		      }
+		    }
+		  },
+		  "take_over": {
+		    "enabled": true
+		  },
+		  "type": "filestream"
+		}
+		`,
+		},
+
+		"parsers are added after json and multiine": {
+			logYamlCfg: `
+id: foo
+paths:
+  - /tmp/foo
+json.expand_keys: true
+multiline.type: count
+parsers:
+  - container: ~
+`,
+			expectedJsonCfg: `
+		{
+		  "file_identity": {
+		    "native": null
+		  },
+		  "id": "foo",
+          "parsers": [
+            {
+              "multiline": {
+                "type": "count"
+              }
+            },
+            {
+              "ndjson": {
+                "expand_keys": true,
+                "target": "json"
+              }
+            },
+            {
+              "container": null
+            }
+          ],
+		  "paths": [
+		    "/tmp/foo"
+		  ],
+		  "prospector": {
+		    "scanner": {
+		      "fingerprint": {
+		        "enabled": false
+		      }
+		    }
+		  },
+		  "take_over": {
+		    "enabled": true
+		  },
+		  "type": "filestream"
+		}
+		`,
+		},
+		"empty values are ignored for all types": {
+			logYamlCfg: `
+# Bool
+close_eof:
+# String
+close_timeout:
+# Shared array
+paths:
+# Bool
+recursive_glob.enabled:
+# Constant
+tail_files:
+# Object
+json:
+multiline:
+`,
+			expectedJsonCfg: `
+{
+  "file_identity": {
+    "native": null
+  },
+  "prospector": {
+    "scanner": {
+      "fingerprint": {
+        "enabled": false
+      }
+    }
+  },
```

Comment:
I think preserving the native file identity would not help our users much.

If `native` is set explicitly, we should preserve it. When the default file identity is used in `log`, we should use the default file identity in `filestream` – `fingerprint`. If I remember correctly, we already support the conversion from `native` to `fingerprint`, however I don't know how it's affected by the enabled `take_over` at the same time.

I don't think we should pursue having identical configuration here, but rather the best configuration for our users when they switch. As long as we have a safe migration path.